### ISSUE TITLE
Rancher CLI login with username/password 

### ISSUE THREAD

**Issue by Tejeev** [rancher/rancher#22254]
**What kind of request is this (question/bug/enhancement/feature request):**
Enhancement

**Steps to reproduce (least amount of steps as possible):**
Login requires secret/token, but it would be nicer to be able to login with a user/pass combo

**Environment information**
- Rancher CLI version: v2.2.0

**Comment by mrproper** [comment]
https://github.com/ibrokethecloud/kubectl-rancher this already does it to get a token
Im not really sure why this feature is missing in the cli. at the moment the only way to get a token is to log into the UI. considering how often tokens expire, this is kinda ordinary

**Comment by mrproper** [comment]
It should also take into account logging in with an IDP provider (okta etc) similar to how the token auth stuff prompts you

**Comment by maxsokolovsky** [comment]
The CLI does have the `token` subcommand, which allows you to authenticate with username and password.
For example, for local authentication, you can use it like this: `rancher token --server {RANCHER_SERVER_URL} --user {USER_ID}`.

Can we close this issue?

**Comment by aadhik** [comment]
Have configured rancher with OKTA as auth provider . Now when running kubectl get pods , it displays a url to  login .Unless you login to the url from browser the token is not created . 
Considering CICD , there would not  interaction involving browser , Can login to extrenal auth provider be done as part of cli

**Comment by maxsokolovsky** [comment]
@aadhik, yes, you can login with an external auth provider via the CLI. The `token` subcommand has the `auth-provider` flag.
But the CLI does not yet support all external auth provider that the API supports. Azure AD OAuth is an example.

**Comment by aadhik** [comment]
@maxsokolovsky - the issue is not with the aut-provider , here is where an operational issue comes 


$ k get pods
https://https://test123.rancher.cloud//v3-public/authProviders
Enter auth provider
0 - localProvider
1 - oktaProvider: 1

Login to Rancher Server at https://test123.rancher.cloud/login?requestId=qfxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&responseType=kubeconfig


AS you can see above selecting okta results in a url  (and the cli prompt goes to wait), this url i have to open in browser and login to it . Once logged in the command line prompt returns back creating rancher cache for further commands and runs fine .


The requirement is , how to pass user/password in cli  ( I can set auth provider to okta to avoid the prompt for it )  so that the  automations can be achieved

**Comment by aadhik** [comment]
@maxsokolovsky  - just to add to above, have disable automatic kubeconfig token generation and configure the expiry time 
(TTL) for kubeconfig tokens. This is done so that if someones leaves organisation , could no longer use token
https://www.suse.com/support/kb/doc/?id=000020021

**Comment by aadhik** [comment]
@maxsokolovsky - even the rancher token cli does not give option to pass password 

rancher token --server=https://test123.rancher.cloud/ --cluster test-ran  --user aadh   --auth-provider=oktaProvider

This also  returns a login url , to which I have to login on browser , then the prompt returns with kubeconfig generated value
.
The use case was , the jenkins , we able to run the command and fetch the token then move on to do dpeloyments , My jenkins resides outside and is a standlone server. Hence the request flow

**Comment by aadhik** [comment]
@maxsokolovsky - for rancher token  cli , is there a way to pass username and password as commnd ? instead doing it interactively

**Comment by maussnma** [comment]
just curious if theres any news? the issiue is close to 5 Years old, and to be honest i cant still log in with username/passwort or generate / request an token.

**Comment by mixxor** [comment]
You can still write a script which does this, but only works with Local Users.

`temp_token=$(curl -s "$RANCHER_URL/v3-public/localProviders/local?action=login" \
        -H 'content-type: application/json' \
        --data-binary "{\"username\":\"$username\",\"password\":\"$password\"}" \
        --insecure | jq -r '.token')

    echo "rancher login --token "$temp_token" --skip-verify "$RANCHER_URL" >/dev/null"`
